src/core_plugins/console/public/src/input.js

```diff
@@ -178,8 +178,13 @@ export function initializeInput($el, $actionsEl, $copyAsCurlEl, output) {
 
             var warnings = xhr.getResponseHeader("warning");
             if (warnings) {
-              warnings = _.map(warnings.split(", "), function (warning) {
-                  return "#! Deprecation: " + warning;
+              // pattern for valid warning header
+              var re = /\d{3} [^ ]+ \"([^"]*)\"( \"[^"]*\")/
+              // split on any comma that is followed by an even number of quotes
+              warnings = _.map(warnings.split(/, (?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)/), function (warning) {
+                var match = re.exec(warning)
```

Comment:
This is a good point. The warning headers produced by Elasticsearch should [always match this pattern](https://github.com/elastic/elasticsearch/pull/23275/files#diff-a2f8f23221b079e15cfc451728820f92R136) but I think your suggestion to err on the side of safety is a good one.